.github/workflows/build-artifacts.yaml

```diff
@@ -0,0 +1,74 @@
+name: Release Build
+
+on:
+  push:
+    paths-ignore:
+      - "*.md"
+      - "compose.*"
+    branches:
+      - "main"
+
+env: 
+  CARGO_TERM_COLOR: always
+
+  CARGO_TARGET_AARCH64_UNKNOWN_LINUX_MUSL_LINKER: aarch64-linux-gnu-gcc
+  CC_aarch64_unknown_linux_musl: aarch64-linux-gnu-gcc
+  CARGO_TARGET_ARMV7_UNKNOWN_LINUX_MUSLEABIHF_LINKER: arm-linux-gnueabihf-gcc
+  CC_armv7_unknown_linux_musleabihf: arm-linux-gnueabihf-gcc
+
+jobs:
+  build:
+    name: Rust project - latest
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        target:
+          - x86_64-unknown-linux-musl
+          - aarch64-unknown-linux-musl
+          - armv7-unknown-linux-musleabihf
+    steps:
+      - uses: actions/checkout@v4
+      
+      - uses: actions-rust-lang/setup-rust-toolchain@v1
+        with:
+          target: ${{ matrix.target }}
+      
+      - if: matrix.target == 'x86_64-unknown-linux-musl'
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y --no-install-recommends musl-tools
+ 
+      - if: matrix.target == 'armv7-unknown-linux-musleabihf'
+        run: |
+          sudo apt update
+          sudo apt install -y gcc-arm-linux-gnueabihf musl-tools
+
+      - if: matrix.target == 'aarch64-unknown-linux-musl'
+        run: |
+          sudo apt update
+          sudo apt install -y gcc-aarch64-linux-gnu musl-tools
+
+      # Using DATE-TIME for release name
+      - name: Date
+        id: date
+        run: echo "DATE=$(date +%Y%m%d-%H%M)" >> "$GITHUB_OUTPUT"
+
+      # In case app version is prefered
+      # - name: Versions
+      #   id: version
+      #   run: echo "VERSION=$(cargo metadata --format-version 1 --no-deps | jq .packages[0].version -r | sed 's/^/v/')" >> "$GITHUB_OUTPUT"
```

Comment:
I think we'd be better off using this especially because other packaging solutions may use the version already to identify new releases